Userland/Utilities/strace.cpp

```diff
@@ -36,24 +137,328 @@ static void handle_sigint(int)
     }
 }
 
-static String format_syscall(syscall_arg_t syscall_index, syscall_arg_t arg1, syscall_arg_t arg2, syscall_arg_t arg3, syscall_arg_t res)
+static void copy_from_process(const void* source_p, void* target_p, size_t size)
+{
+    const char* source = static_cast<const char*>(source_p);
+    char* target = static_cast<char*>(target_p);
+    for (size_t i = 0; i != ceil_div(size, sizeof(int)); i++) {
+        *((int*)target) = ptrace(PT_PEEK, g_pid, const_cast<void*>(static_cast<const void*>(source)), 0);
```

Comment:
This breaks aliasing rules, 1b -> 4b.
Provided that you use the `Bytes` signature above:
```suggestion
        int value = ptrace(PT_PEEK, g_pid, const_cast<char*>(source), 0);
        target.overwrite(0, &value, sizeof(value));
```

Reply:
I ended up doing something similar indeed, had to be a bit more careful with the offset and size given to `overwrite` but it's working :)